### DISCUSSION TITLE
Konkrete Bedürfnisse bezüglich Sandboxes

### DISCUSSION THREAD

**Discussion by rolfrauschenbach** [e-id-admin/governance-sounding-board#37] [General]
Aus nachvollziehbaren Gründen wird immer wieder die Forderung nach Sandboxes an das Projektteam formuliert. Ziel ist es, solche Sandboxes so schnell wie möglich anbieten zu können. Gerne möchten wir hier erfahren, welche konkreten Bedürfnisse hierbei im Vordergrund stehen:

- Welche Use-cases sollen via Sandbox getestet werden?
- Welche technischen Fragen sollen via Sandbox geklärt werden?
- Welche organisatorischen Probleme sollen via Sandbox adressiert werden?
- Welche Governance-Regeln sollen via Sandbox erprobt werden?
- Welche Kapazitäten sollte die Sandbox aufweisen?
- Welchen Support zu der Sandbox wird erwartet?
- etc.

Je konkreter die Bedürfnisse von der Community formuliert werden, desto eher können diese auch befriedigt werden. Danke.

**Comment by StephaneMingot** [comment]
Im Rahmen der SSI-Initiative (siehe https://www.adnovum.com/de/blog/mit-repräsentativen-use-cases-das-potenzial-von-self-sovereign-identity-ausloten) würden wir gerne die drei bereits implementierten Use Case auf die Sandbox portieren. Das Hauptziel dabei soll die Interoperabilität mit anderen Use Cases zu testen bzw. zu ermöglichen. Z.B. Nutzung der Wohnsitzbestätigung von Kanton AG (Use Case 2 der Initiative) für den Use Case "ÖV Partner-GA", der beim Partizipations-Meeting vom 1. September präsentiert wurde, oder die Nutzung der Wohnsitzbestätigung der Stadt St. Gallen für die Immatrikulation eines Fahrzeugs in cardossier (Use Case 3 der Initiative).

**Comment by mloepfe** [comment]
Wir wollen, wie Stéphane vorher geschrieben hat, für unsere diversen PoCs, Demos und Piloten eine VDR (Ledger), die alle verwenden können. Heute sind diese Anwendungen auf den verschiedensten Ledgern verteilt, was die Interoperabilität, resp. die gegenseitige Verwendung von Credentials erschwert oder verunmöglicht.

Diese Sandbox soll keinen produktiven Anforderungen (SLA) genügen müssen, aber trotzdem nahe daran sein, was Stabilität und Verfügbarkeit betrifft. D.h. man soll davon ausgehen können, dass wenn man eine Demo macht, die Sandbox auch funktioniert.

Es soll einfach möglich sein eine Endorser DID zu erhalten, und es soll ein Agent als Endorser zur Verfügung stehen, wo man eine Issuer (Trx-Author) DID registrieren kann.

Ich erwarte keinen Support für die Verwendung der Sandbox. Ansonsten wenn was nicht läuft, sollte zu Bürozeiten (best-effort) reagiert werden.

Bzgl Governance kann man sich an den Testnetzen von Sovrin, Findi, IDUnion etc orientieren. So einfach wie möglich halten.

**Comment by rolfrauschenbach** [comment]
Liebe E-ID-Community
Bisher sind erst zwei Voten zu konkreten Bedürfnissen bezüglich Sandboxes eingegangen. Aufgrund von verschiedenen Wortmeldungen in anderen Zusammenhängen ist das Projektteam E-ID bisher davon ausgegangen, dass ein grösserer Bedarf vorhanden sei. Daher noch einmal der Aufruf, hier konkrete Bedürfnisse zu formulieren. Danke.

**Reply by ChristophGraf-SWITCH** [reply]
Lieber Rolf und Community
Hier nun mit etwas Verzögerung eine kurze Stellungnahme seitens SWITCH:

Welche Use-cases sollen via Sandbox getestet werden?
 - Wir möchten mit der Sandbox Use Cases unterstützen, die die Themen "Diplome" (Elemente wie Abschlüsse, Transcripts, Skills oder besuchte Kurse) und Nachweis der "Angehörigkeit zu Universitäten" (aktuelle Rollen, wie Studierende, Forscher, Lehre oder Administration) tangieren
 - Die Sandbox soll insbesondere Demos mit "echten" Komponenten ermöglichen (Wallets, Registry, etc.) und mit "echten" Issuern und Verifiern, die den künftigen Einsatz von SSI-Techniken erlebbar machen. In der Startphase werden wir vorwiegend mit vordefinierten Personas arbeiten, der Einsatz echter Nutzerdaten wird wohl erst später folgen.
 - Wir erwarten explizit nicht, dass die Sandbox die Wahl der künftigen Infrastruktur präjudiziert. Insbesondere soll es möglich sein Designentscheide für die noch aufzubauende Produktivplattform vom Feedback der Sandbox abhängig zu machen, und dass Entwicklungen z.B. im EU-Raum ebenfalls noch einfliessen können.
 - Wir möchten, dass möglichst viele der laufenden oder angedachten Use Cases über diese Sandbox abgewickelt werden, damit auch sektorübergreifende Use Cases gefördert werden und (mit Hilfe des VC-Credential-Katalogs von DIDAS) auch Use-Case-übergreifende Einsatzszenarien möglich werden.

Welche technischen Fragen sollen via Sandbox geklärt werden?
 - Es soll erlebbar aufgezeigt werden, wie eine mögliche SSI-Umgebung aussehen könnte, wie die verschiedenen Stakeholder in ihren Rollen darin integriert werden, wie man als Nutzer damit interagiert und wieviel Usability man aktuell dabei erwarten kann.
 
Welche organisatorischen Probleme sollen via Sandbox adressiert werden?
 - Demos helfen die organisatorischen Herausforderungen in den verschiedenen Rollen zu erkennen und anzugehen, z.B. welche Prozesse oder Integrationen seitens Issuer oder Verifier aufgebaut und betrieben werden müssen.

Welche Governance-Regeln sollen via Sandbox erprobt werden?
 - Demos helfen die Governance-Herausforderungen in den verschiedenen Rollen zu erkennen und anzugehen, z.B. welche Vertrauensprozesse etabliert werden müssen, welche Rolle darin Sektororganisationen einnehmen sollen oder wie (und welche) Standardisierungsprozesse gestaltet werden sollen.

Welche Kapazitäten sollte die Sandbox aufweisen?
 - Wir erwarten zu Beginn keine Use Cases mit grossen Volumina. Aktuell steht die Funktionalität im Vordergrund.

Welchen Support zu der Sandbox wird erwartet?
 - Zugang zu den benötigten Komponenten wie bereitgestellte Wallet(s), Anforderungen an Wallets Dritter, Integration Guides und Tools (vorzugsweise FOSS) für Issuer und Verifier
 - Wir erwarten in der Startphase unkomplizierten, niederschwelligen "best effort" Support der Sandbox-Betreiber zuhanden der UseCase-Betreiber.

**Comment by Vitus-SBB** [comment]
Mit dem zukünftigen E-ID-Ökosystem betreten wir nicht nur technisch, sondern vor allem auch organisatorisch Neuland. Vieles kann vorausgedacht und -geplant werden. Mindestens ebenso viele Fragen und Probleme, aber auch die Lösungen dazu werden sich erst stellen, wenn wir die ersten Schritte effektiv machen. 
In einem PoC wie demjenigen mit der digitalen Wohnsitzbestätigung konnten zwar viele Erkenntnisse gewonnen werden (u.a. dass wir ein wirklich relevantes Kundenproblem auf kundenfreundliche Art lösen können!). Viele Workarounds verhindern aber, dass wir damit in einen Piloten gehen können oder dass man dieselben Verifiable Credentials in ganz unterschiedlichen use cases mit unterschiedlichen Wallets, Issuers und Verifiern live testen und erleben kann. Letzteres ist aber gerade der Kern des zukünftigen E-ID-Ökosystems. 
Wir haben uses cases (zB mit einem VC Personalausweis), die wir gerne in einem Pilotsystem in realer Umgebung scharf (im Sinne von BETA) austesten würden. Best effort support während Bürozeiten wäre wahrscheinlich genügend, da die aktuellen Prozesse als Fallbacklösungen ja immer noch vorhanden sind.
Für die Governance sollten wir uns an ähnlichen Unterfangen wie zB IDUnion oder Findy orientieren und allfällige Haftungsansprüche kategorisch ausschliessen.

**Comment by timweing** [comment]
Im Rahmen verschiedener Projekt- und Bachelorarbeiten würden wir von der HSLU gerne die Sandbox nutzen. Eine solche Umgebung würde uns die Gelegenheit geben neben rein internen Experimenten auch die Zusammenarbeit mit anderen Projektpartnern in einem produktionsnahen Umfeld anzugehen. Konkrete Fragestellungen, welche uns beschäftigen, sind: Verbindung SSI-Wallet und Crypto-Wallet für KYC Fragen, Anwendungen von Zero-Knowledge Proof oder Registrierung von Hochschulzertifikaten auf Skill-Level.
 
Die Sandbox muss keine SLAs erfüllen, sollte aber eine Uptime von ca. 90-95% haben.
Wie This anmerkte, sollte es einfach sein als Issuer aufgenommen zu werden. 

Um gegenseitig von den Use Cases und Demos lernen zu können, wäre es hervorragend, wenn die Teilnehmenden ihren Source Code auf Github zur Verfügung stellen können und es zu einigen Basisanwendungen auch eine Beschreibung gibt.

**Comment by laurentloup** [comment]
Dear all, 
As mentioned by This, a very light SLA and governance would be necessary to ensure that we can run demos on that sandbox (keep it simple). The objective is indeed to run interop tests and start to experience level of ambition 3 with multiple interactions and combination of credentials. Also, the objective would be to start the normalisation process of credentials (eg residency certificate with communes ,...). Regarding the technological aspects, I would recommend to wait a couple of weeks until the EU commission provides their reference wallet and roadmap. Indeed, it would be good to make everything possible from the beginning to be compatible with what member states will implement in the near future (VC format, signatures, ledger type, .... ). Finally, we would need an alignment with BIT and BJ regarding the timelines and availability to the larger community. If they intend to build a network and make it available to the community in the next few months, the sandbox is probably not necessary. A clear positioning from them would help in that case ...

**Reply by ChristophGraf-SWITCH** [reply]
> Regarding the technological aspects, I would recommend to wait a couple of weeks until the EU commission provides their reference wallet and roadmap.

Alignment with the approaches taken in the EU is really important. But despite this, my preference is not to delay our "Swiss Sandbox" and to start as soon as possible with a sandbox capable to run our use cases. I trust our colleagues at the BIT to exactly deliver such a tool.
When it comes to designing the production platform, we will revisit the international situation - and add our own experiences we gained with our "Swiss Sandbox". 
And BTW, we will certainly get the opportunity to talk about our experiences not just in the national context, but also to discuss them with our peers internationally - it's not just about copying, but also about contributing.

**Comment by sgw** [comment]
Dear all,

We are currently scoping an SSI PoC in the domain of memberships and tickets for leisure activities (e.g. sports). The PoC scoping is done for the Kanton of Thurgau and is performed with the support of Acrea, a consulting firm in the field of digitization.

The implementation of this PoC is planned for 2023 and is expected to somehow build upon the federal infrastructure. Therefore, using the sandbox will be an obvious approach and key pre-requisite for conducting the PoC as planned.

Many of the sandbox requirements and expectations already have been mentioned in this thread.

What follows is a summary of the requirements we currently see for the mentioned PoC:

	• The provided functionality should cover:
		○ Registration of issuers
		○ Issuing of VCs for a designated issuer
		○ Revocation of VCs
		○ Invalidation of VCs
		○ Support for verification of VCs
	• The functionality as described above should be accessible via Web clients (prio 1) and APIs (prio 2)
	• SSI Wallet bundled with the sandbox or support for a 3rd party wallet (e.g. ESATUS)
	• Issuing a 'sandbox e-id' (according to the latest spec draft) should be possible in a simple way.
	• When it comes to governance rules, it shall be possible to restrict issuance of specific types of VCs to explicitly nominated issuers
	• The combination of use cases should be possible, e.g. issue a VC based on a VC that is already in the users wallet
	• It should be possible to instantiate an arbitrary number of sandbox instances (for experimenting, testing, …)
